frontend/mainwindow.cpp

```diff
@@ -62,9 +93,13 @@ void MainWindow::fillInstruments() {
   foreach (const QString instrument, instruments) {
     ui->instrumentsBox->addItem(instrument);
   }
+  connect(ui->instrumentsBox, SIGNAL(currentTextChanged(const QString)), this,
+          SLOT(instrumentsBoxChange(const QString)));
 }
 
-void MainWindow::on_instrumentsBox_currentTextChanged(const QString &arg1) {
+void MainWindow::instrumentsBoxChange(const QString &arg1) {
+  QSettings settings;
+  settings.setValue("recentInstrument", arg1);
```

Comment:
I would argue that it's probably better practice to rewrite all data in the `QSettings` object on close of the frontend's main window  - you can override `QMainWindow::closeEvent(QCloseEvent *event)` and do it there.